openmmtools/forcefactories.py

```diff
@@ -174,6 +174,240 @@ def restrain_atoms(thermodynamic_state, sampler_state, restrained_atoms, sigma=3
+    #new_force.setForceGroup(nonbonded_force.getForceGroup())
+    new_force.setNonbondedMethod(nonbonded_force.getNonbondedMethod())
+    #new_force.setPMEParameters(*nonbonded_force.getPMEParameters())
+    #new_force.setReactionFieldDielectric(nonbonded_force.getReactionFieldDielectric())
+    #new_force.setReciprocalSpaceForceGroup(nonbonded_force.getReciprocalSpaceForceGroup())
+    new_force.setUseSwitchingFunction(nonbonded_force.getUseSwitchingFunction())
+    new_force.setSwitchingDistance(nonbonded_force.getSwitchingDistance())
+    new_force.setUseLongRangeCorrection(nonbonded_force.getUseDispersionCorrection())
+
+
+    # now add all the particle parameters
+    num_particles = nonbonded_force.getNumParticles()
+    for i in range(num_particles):
+        new_force.addParticle(nonbonded_force.getParticleParameters(i))
+
+    # now add all the exceptions? # TODO: check if we want to do this...
+    #    for exception_index in range(nonbonded_force.getNumExceptions()):
+    #        new_force.addException(nonbonded_force.getExceptionParameters(exception_index))
+
+    # TODO: There's probably a cleaner, more Pythonic way to do this
+    # (this was the most obvious way, but may not work in the future if the OpenMM API changes)
+
+    return new_force
+
+
+def split_nb_using_interaction_groups(system, md_topology):
+    """Construct a copy of system where its nonbonded force has been replaced
+    with two nonbonded forces, one using an interaction group restricted to
+    water-water interactions, and the other using interaction groups
+    restricted to water-solute and solute-solute interactions. Water-water
+    interactions are in force group 0, and all other interactions are in force
+    group 1.
+    """
+
+    # create a copy of the original system: only touch this
+    new_system = copy.deepcopy(system)
+
+    # find the default nonbonded force
+    force_index, nb_force = forces.find_forces(new_system, openmm.NonbondedForce, only_one=True)
+    # create copies for each interaction. Only half in solvent/solvent and solute/solute as we double-count.
+    nb_only_solvent_solvent = clone_nonbonded_parameters(nb_force, energy_prefactor='0.5*')
+    nb_solute = clone_nonbonded_parameters(nb_force)
+
+    # NOTE: these need to be python ints -- not np.int64s -- when passing to addInteractionGroup later!
+    solvent_indices = list(map(int, md_topology.select('water')))
+    solute_indices = list(map(int, md_topology.select('not water')))
+
+    #all_indices = list(range(md_topology.n_atoms))
+
+    nb_only_solvent_solvent.addInteractionGroup(set1=solvent_indices, set2=solvent_indices)
+
+    nb_solute.addInteractionGroup(set1=solute_indices, set2=solute_indices)
+    nb_solute.addInteractionGroup(set1=solute_indices, set2=solvent_indices)
+
+    #nb_solute.addInteractionGroup(set1=solute_indices, set2=all_indices)
+
+    # remove original force, add new forces
+    new_system.removeForce(force_index)
```

Comment:
I think you want to leave in the original force so you don't have to copy all the exceptions (nonzero 1-4 interactions) to a CustomBondForce.

Reply:
Okay!